### PR TITLE
Dropdown component

### PR DESCRIPTION
**Pull request by Efekahya**
resim

https://www.figma.com/file/SBPCNjmePIwJCzqqhXF9Y8/Rick-and-Morty---Digieggs-Assessment?node-id=1014%3A944

### CODE REVIEW

File: src/components/Dropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { IDropdownProps } from "../../types/interfaces";
+import { ReactComponent as Arrow } from "../../assets/arrowDown.svg";
+export default function Dropdown({ items, selected }: IDropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const handleDropdown = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setIsOpen(!isOpen);
+  };
+  return (
+    <div className={"container"}>
```
Comment (iremlaya) [inline src/components/Dropdown/index.tsx]:
```suggestion
    <div className="container">
```

File: src/components/Dropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import "./styles.scss";
+import { IDropdownProps } from "../../types/interfaces";
+import { ReactComponent as Arrow } from "../../assets/arrowDown.svg";
```
Comment (iremlaya) [inline src/components/Dropdown/index.tsx:3]: should be a new line below

File: src/components/Dropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+export default function Dropdown({ items, selected }: IDropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const handleDropdown = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setIsOpen(!isOpen);
+  };
```
Comment (iremlaya) [inline src/components/Dropdown/index.tsx:11]: maybe a new line below as well, for readability

File: src/components/Dropdown/styles.scss [hunk 4]
```diff
@@ -0,0 +1,82 @@
+.container{
+  width: 90%;
```
Comment (iremlaya) [inline src/components/Dropdown/styles.scss]: why 90%

File: src/components/Dropdown/styles.scss [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    align-items: center;
+    width: 100%;
+    height: 56px;
+    padding: 16px;
+    margin-bottom:8px;
+    background: #F5F5F5;
+    border-radius: 8px;
+    box-sizing: border-box;
+    &:active{
+        background: #BBBBBB;
```
Comment (iremlaya) [inline src/components/Dropdown/styles.scss]: I'm not sure if you have a theme file, but try to declare colors as theme variables

File: src/components/Dropdown/styles.scss [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    height: 100%;
+    position: absolute ;
+}
+
+.psuedo {
+    position: relative;
+    width: 100%;
+}
+.open{
+
```
Comment (iremlaya) [inline src/components/Dropdown/styles.scss:49]: unnecessary new line

File: src/components/Dropdown/styles.scss [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    opacity: 1;
+    transition: opacity 0.2s ease-in-out;
+    display: flex;
+}
+
+.dropdown-item{
+all: unset;
```
Comment (iremlaya) [inline src/components/Dropdown/styles.scss]: add space to this section

**Review by iremlaya — CHANGES_REQUESTED** [review]

File: src/components/Dropdown/index.tsx (hunk 1 above)
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
    <div className="container">
```

File: src/components/Dropdown/index.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import "./styles.scss";
+import { IDropdownProps } from "../../types/interfaces";
+import { ReactComponent as Arrow } from "../../assets/arrowDown.svg";
+export default function Dropdown({ items, selected }: IDropdownProps) {
+  const [isOpen, setIsOpen] = useState(false);
+  const handleDropdown = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setIsOpen(!isOpen);
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
    setIsOpen(oldValue => !oldValue);
```

File: src/components/Dropdown/index.tsx (hunk 2 above)
Comment (atvdi) [inline src/components/Dropdown/index.tsx:3]: Nothing serious but I would add a newline after imports.
```suggestion
import { ReactComponent as Arrow } from "../../assets/arrowDown.svg";

```

File: src/components/Dropdown/index.tsx [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  };
+  return (
+    <div className={"container"}>
+      <button className="dropdown" onClick={handleDropdown}>
+        {selected}
+        <Arrow className={"arrow " + (isOpen ? "down" : "")} />
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
        <Arrow className={`arrow ${isOpen ? "down" : ""}`} />
```

File: src/components/Dropdown/index.tsx [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+      </button>
+      <div className="psuedo">
+        <div className={"itemsContainer " + (isOpen ? "open" : "")}>
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
        <div className={`itemsContainer ${isOpen ? "open" : ""}`}>
```

File: src/components/Dropdown/index.tsx [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+          {items.map(item => (
+            <div className="item">
+              <button
+                key={item.value}
+                value={item.value}
+                className={"dropdown-item "}
+                onClick={event => {
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]: Let's move this arrow function to outside.

File: src/components/Dropdown/index.tsx [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+        {selected}
+        <Arrow className={"arrow " + (isOpen ? "down" : "")} />
+      </button>
+      <div className="psuedo">
+        <div className={"itemsContainer " + (isOpen ? "open" : "")}>
+          {items.map(item => (
+            <div className="item">
+              <button
+                key={item.value}
+                value={item.value}
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
                key={value}
                value={value}
```

File: src/components/Dropdown/index.tsx [hunk 13]
```diff
@@ -0,0 +1,38 @@
...
+  };
+  return (
+    <div className={"container"}>
+      <button className="dropdown" onClick={handleDropdown}>
+        {selected}
+        <Arrow className={"arrow " + (isOpen ? "down" : "")} />
+      </button>
+      <div className="psuedo">
+        <div className={"itemsContainer " + (isOpen ? "open" : "")}>
+          {items.map(item => (
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
          {items.map(({value, label, handleClick})=> (
```

File: src/components/Dropdown/index.tsx [hunk 14]
```diff
@@ -0,0 +1,38 @@
...
+              <button
+                key={item.value}
+                value={item.value}
+                className={"dropdown-item "}
+                onClick={event => {
+                  item.handleClick(event);
+                  setIsOpen(false);
+                }}
+              >
+                {item.label}
```
Comment (atvdi) [inline src/components/Dropdown/index.tsx]:
```suggestion
                 handleClick(event);
                  setIsOpen(false);
                }}
              >
                {label}
```

File: src/components/Dropdown/styles.scss [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+    z-index: 10;
+    display: flex;
+    flex-direction: column;
+    justify-content: flex-start;
+    align-items: flex-start;
+    background: #F5F5F5;
+    opacity: 0;
+    width: 100%;
+    height: 100%;
+    position: absolute ;
```
Comment (atvdi) [inline src/components/Dropdown/styles.scss]: You should use a formatter for css files.
```suggestion
    position: absolute;
```

File: src/components/Dropdown/styles.scss (hunk 6 above)
Comment (atvdi) [inline src/components/Dropdown/styles.scss:49]: Unnecessary newline

File: src/components/Dropdown/styles.scss [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+    background: #BBBBBB;
+}
+}
+.item{
+    position: relative;
+    width: 100%;
+    height: 56px;
+    box-sizing: border-box;
+    background: #F5F5F5;
+
```
Comment (atvdi) [inline src/components/Dropdown/styles.scss]: Unnecessary newline

**Review by atvdi — CHANGES_REQUESTED** [review]

File: src/components/Dropdown/index.tsx (hunk 8 above)
Comment (Efekahya) [reply src/components/Dropdown/index.tsx]: What is the reason for doing it this way?

File: src/components/Dropdown/index.tsx (hunk 9 above)
Comment (Efekahya) [reply src/components/Dropdown/index.tsx]: What is the reason for doing it this way?

File: src/components/Dropdown/styles.scss [hunk 17]
```diff
@@ -0,0 +1,87 @@
...
+    background: $gray;
+    border-radius: 8px;
+    box-sizing: border-box;
+
+    &:active {
+        background: $black ;
+    }
+
+    &:hover {
+        background: #E6E6E6;
```
Comment (atvdi) [inline src/components/Dropdown/styles.scss]: Why this is not a variable?

File: src/components/Navbar/styles.scss [hunk 18]
```diff
@@ -1,4 +1,4 @@
-.navbar {
+.dropdown--navbar {
```
Comment (atvdi) [inline src/components/Navbar/styles.scss:1]: Why are you using double dashes in class names? Is there any special reason?

**Review by atvdi — COMMENTED** [review]
Just a couple of comments. Good work.

File: src/components/Dropdown/styles.scss (hunk 17 above)
Comment (Efekahya) [reply src/components/Dropdown/styles.scss]: this color was used in one place only. So there was no reason for me to make this a variable. Should I make this a variable anyway?

File: src/components/Navbar/styles.scss (hunk 18 above)
Comment (Efekahya) [reply src/components/Navbar/styles.scss:1]: There is no particular reason. I think it looks more clear but I will change it on later commits.

File: src/components/Dropdown/styles.scss (hunk 17 above)
Comment (atvdi) [reply src/components/Dropdown/styles.scss]: For the code integrity, I would recommend to do so.

**Pull request merged**